app/uk/gov/hmrc/helptosavefrontend/util/EmailVerificationParams.scala

```diff
@@ -40,8 +40,9 @@ object EmailVerificationParams {
     Try(new String(decoder.decode(base64))).flatMap(crypto.decrypt) match {
       case Failure(_) ⇒ None
       case Success(decrypted) ⇒
-        val params = decrypted.split("§")
-        if (params.length === 2) Some(EmailVerificationParams(params(0), params(1))) else None
+        val nino = decrypted.substring(0, decrypted.indexOf('#'))
```

Comment:
if there isn't a `#` in the string, `.substring` is going to throw a `java.lang.StringIndexOutOfBoundsException`. But this should be OK since we are wrapping this whole decode logic in a `Try` so it should get picked up